### PR TITLE
Clean headers from A-C

### PR DESCRIPTION
**Pull request by brianlball**
Clean Headers from A-C

### CODE REVIEW

File: src/EnergyPlus/Boilers.hh [hunk 1]
```diff
@@ -197,15 +182,12 @@ namespace Boilers {
...
 {
-    int numBoilers = 0;
     bool getBoilerInputFlag = true;
     Array1D<Boilers::BoilerSpecs> Boiler;
 
     void clear_state() override
     {
-        this->numBoilers = 0;
-        this->getBoilerInputFlag = true;
-        this->Boiler.deallocate();
```
Comment (amirroth) [inline src/EnergyPlus/Boilers.hh:210]: Why do we no longer have to call `deallocate()`? Did we not have to call it to begin with?

File: src/EnergyPlus/ConvectionCoefficients.hh [hunk 2]
```diff
@@ -75,231 +75,149 @@ namespace ConvectionCoefficients {
...
-              MixedBuoyOpposingFlowWallEqNum(ConvectionConstants::HcInt_BeausoleilMorrisonMixedOppossingWall),
-              MixedBuoyOpposingFlowWallUserCurveNum(0), MixedStableFloorEqNum(ConvectionConstants::HcInt_BeausoleilMorrisonMixedStableFloor),
-              MixedStableFloorUserCurveNum(0), MixedUnstableFloorEqNum(ConvectionConstants::HcInt_BeausoleilMorrisonMixedUnstableFloor),
-              MixedUnstableFloorUserCurveNum(0), MixedStableCeilingEqNum(ConvectionConstants::HcInt_BeausoleilMorrisonMixedStableCeiling),
-              MixedStableCeilingUserCurveNum(0), MixedUnstableCeilingEqNum(ConvectionConstants::HcInt_BeausoleilMorrisonMixedUnstableCeiling),
-              MixedUnstableCeilingUserCurveNum(0), MixedWindowsEqNum(ConvectionConstants::HcInt_GoldsteinNovoselacCeilingDiffuserWindow),
-              MixedWindowsUserCurveNum(0)
-        {
-        }
+        int SimpleBuoyVertWallEqNum = ConvectionConstants::HcInt_FohannoPolidoriVerticalWall; // InConvClass_A3_VertWalls
```
Comment (amirroth) [inline src/EnergyPlus/ConvectionCoefficients.hh:101]: Straight up one of the most ridiculous data structures in EnergyPlus.

File: src/EnergyPlus/CurveManager.hh [hunk 3]
```diff
@@ -119,123 +119,101 @@ namespace CurveManager {
...
+        Real64 CoeffA17 = 0.0;
+        Real64 CoeffA18 = 0.0;
+        Real64 CoeffA19 = 0.0;
+        Real64 CoeffA20 = 0.0;
+        Real64 CoeffA21 = 0.0;
+        Real64 CoeffA22 = 0.0;
+        Real64 CoeffA23 = 0.0;
+        Real64 CoeffA24 = 0.0;
+        Real64 CoeffA25 = 0.0;
+        Real64 CoeffA26 = 0.0;
```
Comment (amirroth) [inline src/EnergyPlus/CurveManager.hh:148]: What programming was like before arrays were invented.

File: src/EnergyPlus/ConvectionCoefficients.hh [hunk 4]
```diff
@@ -316,50 +234,30 @@ namespace ConvectionCoefficients {
...
-                                       Real64 const Xmax,
-                                       Real64 const Xmin,
-                                       Real64 const Area,
-                                       Real64 const Perimeter,
-                                       Real64 const Height)
-            : AzimuthRangeLow(AzimuthRangeLow), AzimuthRangeHi(AzimuthRangeHi), Zmax(Zmax), Zmin(Zmin), Ymax(Ymax), Ymin(Ymin), Xmax(Xmax),
-              Xmin(Xmin), Area(Area), Perimeter(Perimeter), Height(Height)
-        {
-        }
+        Real64 AzimuthRangeLow = AzimuthRangeLow;
```
Comment (amirroth) [inline src/EnergyPlus/ConvectionCoefficients.hh]: Ummmmm ... what's this?

File: src/EnergyPlus/ConvectionCoefficients.hh [hunk 5]
```diff
@@ -75,231 +75,149 @@ namespace ConvectionCoefficients {
...
     };
 
     struct BoundingBoxVertStruct
     {
         // Members
-        int SurfNum;
-        int VertNum;
+        int SurfNum = 0;
+        int VertNum = 0;
         Vector Vertex;
```
Comment (amirroth) [inline src/EnergyPlus/ConvectionCoefficients.hh]: Can you not initialize this here too?

File: src/EnergyPlus/ConvectionCoefficients.hh [hunk 6]
```diff
@@ -316,50 +234,30 @@ namespace ConvectionCoefficients {
         BoundingBoxVertStruct XuYdZu; // 7 hi x, low y, hi z
         BoundingBoxVertStruct XuYuZu; // 8 hi x, hi y, hi z
         Array1D<Vector> BoundSurf;    // long edge of roof group bounding surface
```
Comment (amirroth) [inline src/EnergyPlus/ConvectionCoefficients.hh:231]: This could be a `std::array` in which case you could initialize it here.

File: src/EnergyPlus/ConvectionCoefficients.hh (hunk 5 above)
Comment (brianlball) [reply src/EnergyPlus/ConvectionCoefficients.hh]: yeah, this one is easy to change

File: src/EnergyPlus/ConvectionCoefficients.hh (hunk 6 above)
Comment (brianlball) [reply src/EnergyPlus/ConvectionCoefficients.hh:231]: this one is used in https://github.com/NREL/EnergyPlus/blob/CleanHeadersFromAdown/src/EnergyPlus/ConvectionCoefficients.cc#L4515 which will require changing several methods and could get out of hand quickly.  we can add it to the to be continued list

File: src/EnergyPlus/ConvectionCoefficients.hh (hunk 4 above)
Comment (brianlball) [reply src/EnergyPlus/ConvectionCoefficients.hh]: its some kind of voodoo... i'm reverting this conversion back, its too weird

Comment (Myoldmopar) [reply src/EnergyPlus/ConvectionCoefficients.hh]: OK, so in this case, this struct has a member-constructor, where the caller provides all the inputs as arguments and they are applied to the class members.  We should indeed leave the member constructor here...unless it's not being used anywhere in the code (hint: if it compiled without it then it must not be used anywhere).  We are really just trying to clean up the structs that only use default constructors with the initial values listed in the initializer lists.

Comment (amirroth) [reply src/EnergyPlus/ConvectionCoefficients.hh]: I wasn't asking about the member constructor, I was asking about what `Real64 AzimuthRangeLow = AzimuthRangeLow;` is intended to accomplish and actually accomplishes. It looks like it doesn't do anything.

File: src/EnergyPlus/ConvectionCoefficients.hh (hunk 6 above)
Comment (brianlball) [reply src/EnergyPlus/ConvectionCoefficients.hh:231]: i'm making an issue to keep track of all these more involved refactoring changes so they dont get lost. 
 https://github.com/NREL/EnergyPlus/issues/9372

File: src/EnergyPlus/BoilerSteam.hh [hunk 7]
```diff
@@ -70,57 +70,50 @@ namespace BoilerSteam {
...
         // Default Constructor
-        BoilerSpecs()
-            : FuelType(DataGlobalConstants::ResourceType::None), Available(false), ON(false), MissingSetPointErrDone(false), UseLoopSetPoint(false),
-              DesMassFlowRate(0.0), MassFlowRate(0.0), NomCap(0.0), NomCapWasAutoSized(false), NomEffic(0.0), MinPartLoadRat(0.0),
-              MaxPartLoadRat(0.0), OptPartLoadRat(0.0), OperPartLoadRat(0.0), TempUpLimitBoilerOut(0.0), BoilerMaxOperPress(0.0),
-              BoilerPressCheck(0.0), SizFac(0.0), BoilerInletNodeNum(0), BoilerOutletNodeNum(0), FullLoadCoef(3, 0.0), TypeNum(0), plantLoc{},
-              PressErrIndex(0), FluidIndex(0), myFlag(true), myEnvrnFlag(true), FuelUsed(0.0), BoilerLoad(0.0), BoilerEff(0.0),
-              BoilerMassFlowRate(0.0), BoilerOutletTemp(0.0), BoilerEnergy(0.0), FuelConsumed(0.0), BoilerInletTemp(0.0),
-              BoilerFuelTypeForOutputVariable("")
+        BoilerSpecs() : plantLoc{}
```
Comment (amirroth) [inline src/EnergyPlus/BoilerSteam.hh]: Why is this necessary? Why is `plantLoc` default constructor not called automatically?

File: src/EnergyPlus/BoilerSteam.hh [hunk 8]
```diff
@@ -171,15 +164,12 @@ namespace BoilerSteam {
 
 struct BoilerSteamData : BaseGlobalStruct
 {
-    int numBoilers = 0;
     bool getSteamBoilerInput = true;
     Array1D<BoilerSteam::BoilerSpecs> Boiler;
```
Comment (amirroth) [inline src/EnergyPlus/BoilerSteam.hh:163]: There is no reason for a 1D array of anything that isn't `int`, `Real64`, or `bool` to be an `Array1D` rather than an `EPVector`.

File: src/EnergyPlus/BranchInputManager.cc [hunk 9]
```diff
@@ -1051,9 +1051,9 @@ namespace BranchInputManager {
 
         if (state.dataBranchInputManager->GetBranchInputOneTimeFlag) {
             std::string CurrentModuleObject = "Branch";
-            state.dataBranchInputManager->NumOfBranches = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, CurrentModuleObject);
-            if (state.dataBranchInputManager->NumOfBranches > 0) {
-                state.dataBranchInputManager->Branch.allocate(state.dataBranchInputManager->NumOfBranches);
+            int NumOfBranches = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, CurrentModuleObject);
+            if (NumOfBranches > 0) {
+                state.dataBranchInputManager->Branch.allocate(NumOfBranches);
                 for (auto &e : state.dataBranchInputManager->Branch)
```
Comment (amirroth) [inline src/EnergyPlus/BranchInputManager.cc]: Why is this loop necessary?

File: src/EnergyPlus/BranchInputManager.cc [hunk 10]
```diff
@@ -1321,8 +1321,8 @@ namespace BranchInputManager {
 
         ErrFound = false;
         std::string CurrentModuleObject = "BranchList";
-        state.dataBranchInputManager->NumOfBranchLists = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, CurrentModuleObject);
-        state.dataBranchInputManager->BranchList.allocate(state.dataBranchInputManager->NumOfBranchLists);
+        int NumOfBranchLists = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, CurrentModuleObject);
+        state.dataBranchInputManager->BranchList.allocate(NumOfBranchLists);
         for (auto &e : state.dataBranchInputManager->BranchList) {
```
Comment (amirroth) [inline src/EnergyPlus/BranchInputManager.cc]: Why is this loop necessary?

File: src/EnergyPlus/BranchInputManager.cc [hunk 11]
```diff
@@ -2743,7 +2743,7 @@ namespace BranchInputManager {
         }
 
         // Build node names in branches
-        for (Count = 1; Count <= state.dataBranchInputManager->NumOfBranches; ++Count) {
+        for (Count = 1; Count <= (int)state.dataBranchInputManager->Branch.size(); ++Count) {
             BranchNodes(Count).UniqueNodeNames.allocate(state.dataBranchInputManager->Branch(Count).NumOfComponents * 2);
             BranchNodes(Count).UniqueNodeNames = std::string();
```
Comment (amirroth) [inline src/EnergyPlus/BranchInputManager.cc]: You just did an `allocate` on this and now you assign it to a blank string because ... ?

File: src/EnergyPlus/ChilledCeilingPanelSimple.cc [hunk 12]
```diff
@@ -257,18 +257,17 @@ void GetCoolingPanelInput(EnergyPlusData &state)
...
 
     // Count total number of baseboard units
 
-    state.dataChilledCeilingPanelSimple->CoolingPanel.allocate(state.dataChilledCeilingPanelSimple->NumCoolingPanels);
-    state.dataChilledCeilingPanelSimple->CoolingPanelSysNumericFields.allocate(state.dataChilledCeilingPanelSimple->NumCoolingPanels);
-    state.dataChilledCeilingPanelSimple->CheckEquipName.allocate(state.dataChilledCeilingPanelSimple->NumCoolingPanels);
+    state.dataChilledCeilingPanelSimple->CoolingPanel.allocate(NumCoolingPanels);
+    state.dataChilledCeilingPanelSimple->CoolingPanelSysNumericFields.allocate(NumCoolingPanels);
+    state.dataChilledCeilingPanelSimple->CheckEquipName.allocate(NumCoolingPanels);
     state.dataChilledCeilingPanelSimple->CheckEquipName = true;
```
Comment (amirroth) [inline src/EnergyPlus/ChilledCeilingPanelSimple.cc]: Does this assignment operator set all elements of the array to `true`? That is a terrible idiom. Is this a Fortran legacy thing?

**Comment by Myoldmopar** [comment]
GPU diff, otherwise CI is clean here.  I think it's time to start pushing this toward merge.  I need to digest the comments above next.

File: src/EnergyPlus/BoilerSteam.cc [hunk 13]
```diff
@@ -161,10 +161,9 @@ namespace BoilerSteam {
 
         SteamFluidIndex = 0;
         state.dataIPShortCut->cCurrentModuleObject = "Boiler:Steam";
-        state.dataBoilerSteam->numBoilers =
-            state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, state.dataIPShortCut->cCurrentModuleObject);
+        int numBoilers = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, state.dataIPShortCut->cCurrentModuleObject);
```
Comment (Myoldmopar) [inline src/EnergyPlus/BoilerSteam.cc:164]: Yay, every time an unnecessary variable is removed from state and replaced with a local variable, an angel smiles.

File: src/EnergyPlus/BoilerSteam.cc [hunk 14]
```diff
@@ -229,9 +228,9 @@ namespace BoilerSteam {
             thisBoiler.MinPartLoadRat = state.dataIPShortCut->rNumericArgs(5);
             thisBoiler.MaxPartLoadRat = state.dataIPShortCut->rNumericArgs(6);
             thisBoiler.OptPartLoadRat = state.dataIPShortCut->rNumericArgs(7);
-            thisBoiler.FullLoadCoef(1) = state.dataIPShortCut->rNumericArgs(8);
-            thisBoiler.FullLoadCoef(2) = state.dataIPShortCut->rNumericArgs(9);
-            thisBoiler.FullLoadCoef(3) = state.dataIPShortCut->rNumericArgs(10);
+            thisBoiler.FullLoadCoef[0] = state.dataIPShortCut->rNumericArgs(8);
```
Comment (Myoldmopar) [inline src/EnergyPlus/BoilerSteam.cc:231]: Ahhh, zero based indexing and square brackets, the way arrays were meant to be in C++ :smile:

File: src/EnergyPlus/BoilerSteam.hh (hunk 7 above)
Comment (Myoldmopar) [reply src/EnergyPlus/BoilerSteam.hh]: This is not necessary.

File: src/EnergyPlus/BoilerSteam.hh (hunk 8 above)
Comment (Myoldmopar) [reply src/EnergyPlus/BoilerSteam.hh:163]: I agree, we are definitely moving in that direction.  We can go back through these few files and add an extra effort to make sure all the Array1Ds are transitioned.  We were trying to balance the header file cleanup with how many lines we had to change in the cc files.  Changing this to a vector will cause quite a lot of changes, but I'm OK adding that here if you want.

File: src/EnergyPlus/BoilerSteam.hh [hunk 15]
```diff
@@ -171,15 +164,12 @@ namespace BoilerSteam {
...
 
     void clear_state() override
     {
-        this->numBoilers = 0;
-        this->getSteamBoilerInput = true;
-        this->Boiler.deallocate();
+        *this = BoilerSteamData();
```
Comment (Myoldmopar) [inline src/EnergyPlus/BoilerSteam.hh:167]: Yay!  So much DRYer!

File: src/EnergyPlus/Boilers.hh [hunk 16]
```diff
@@ -82,74 +82,64 @@ namespace Boilers {
...
-              ModulatedFlowSetToLoop(false), ModulatedFlowErrDone(false), VolFlowRate(0.0), VolFlowRateWasAutoSized(false), DesMassFlowRate(0.0),
-              MassFlowRate(0.0), SizFac(0.0), BoilerInletNodeNum(0), BoilerOutletNodeNum(0), MinPartLoadRat(0.0), MaxPartLoadRat(0.0),
-              OptPartLoadRat(0.0), OperPartLoadRat(0.0), CurveTempMode(TempMode::NOTSET), EfficiencyCurvePtr(0), TempUpLimitBoilerOut(0.0),
-              ParasiticElecLoad(0.0), EffCurveOutputError(0), EffCurveOutputIndex(0), CalculatedEffError(0), CalculatedEffIndex(0),
-              IsThisSized(false), FaultyBoilerFoulingFlag(false), FaultyBoilerFoulingIndex(0), FaultyBoilerFoulingFactor(1.0), MyEnvrnFlag(true),
-              MyFlag(true), FuelUsed(0.0), ParasiticElecPower(0.0), BoilerLoad(0.0), BoilerMassFlowRate(0.0), BoilerOutletTemp(0.0), BoilerPLR(0.0),
-              BoilerEff(0.0), BoilerEnergy(0.0), FuelConsumed(0.0), BoilerInletTemp(0.0), ParasiticElecConsumption(0.0),
-              BoilerFuelTypeForOutputVariable("")
+        BoilerSpecs() : plantLoc{}
```
Comment (Myoldmopar) [inline src/EnergyPlus/Boilers.hh:142]: This can be removed.

File: src/EnergyPlus/Boilers.hh (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/Boilers.hh:210]: We don't need the deallocate anymore because we are creating a new instance.  For the same reason, we don't need to manually go member-by-member resetting them to their original values anymore.  It's so much nicer.  Once we get all these clear_states cleaned out to the minimal new code, I think we will have a very rapid change where we don't really need the individual clear_state functions anymore.  But we'll see...

File: src/EnergyPlus/BranchInputManager.cc (hunk 9 above)
Comment (Myoldmopar) [reply src/EnergyPlus/BranchInputManager.cc]: It doesn't look like it should be now.  But it's been in there since the C++ translation.  Back in Fortran, right after Branch was allocated, the AssignedLoopName was set to a blank.  I think ....maybe.... it was because in Fortran you couldn't rely on the fixed-length strings to be filled with blank spaces?   Maybe...?

File: src/EnergyPlus/BranchInputManager.cc (hunk 11 above)
Comment (Myoldmopar) [reply src/EnergyPlus/BranchInputManager.cc]: I think these all go back to the original Fortran code, I'm not sure if there is a way we can automatically detect these unnecessary things now...but maybe.  Anytime we reset a string to a blank seems like a good starting point, since that seems like an odd thing to do inside the simulation.  Then we could add some intelligence to snoop just a couple lines before that assignment to see if it detects an allocation of that variable (or any variable?).  Could be a fun excursion.

File: src/EnergyPlus/BranchInputManager.hh [hunk 17]
```diff
@@ -371,24 +319,7 @@ struct BranchInputManagerData : BaseGlobalStruct
...
-        this->GetConnectorListInputFlag = true;
-        this->InvalidBranchDefinitions = false;
-        this->GetBranchInputOneTimeFlag = true;
-        this->BranchList.deallocate();
-        this->Branch.deallocate();
-        this->ConnectorLists.deallocate();
-        this->Splitters.deallocate();
-        this->Mixers.deallocate();
-        this->BComponents.deallocate();
+        *this = BranchInputManagerData();
```
Comment (Myoldmopar) [inline src/EnergyPlus/BranchInputManager.hh:322]: Yay!!!

File: src/EnergyPlus/CTElectricGenerator.hh [hunk 18]
```diff
@@ -69,74 +69,64 @@ namespace CTElectricGenerator {
...
+        Real64 FuelHeatingValue = 0.0;          // Heating Value for Fuel in (kJ/kg)
+        Real64 ElecPowerGenerated = 0.0;        // reporting: power generated (W)
+        Real64 ElecEnergyGenerated = 0.0;       // reporting: power generated (W)
+        Real64 HeatRecMaxTemp = 0.0;            // Max Temp that can be produced in heat recovery
+        int OAInletNode = 0;                    // optional inlet node index pointer for outdoor air for combustion
+        bool MyEnvrnFlag = true;
+        bool MyPlantScanFlag = true;
+        bool MySizeAndNodeInitFlag = true;
+        bool CheckEquipName = true;
+        bool MyFlag = true;
```
Comment (Myoldmopar) [inline src/EnergyPlus/CTElectricGenerator.hh:126]: It is so much nicer to see the initial values listed on the variable declarations, not tucked inside the mess of the initializer list on the constructor.  So much nicer.

File: src/EnergyPlus/CTElectricGenerator.hh [hunk 19]
```diff
@@ -69,74 +69,64 @@ namespace CTElectricGenerator {
...
-              MinPartLoadRat(0.0), MaxPartLoadRat(0.0), OptPartLoadRat(0.0), FuelEnergyUseRate(0.0), FuelEnergy(0.0), PLBasedFuelInputCurve(0),
-              TempBasedFuelInputCurve(0), ExhaustFlow(0.0), ExhaustFlowCurve(0), ExhaustTemp(0.0), PLBasedExhaustTempCurve(0),
-              TempBasedExhaustTempCurve(0), QLubeOilRecovered(0.0), QExhaustRecovered(0.0), QTotalHeatRecovered(0.0), LubeOilEnergyRec(0.0),
-              ExhaustEnergyRec(0.0), TotalHeatEnergyRec(0.0), QLubeOilRecoveredCurve(0), UA(0.0), UACoef(2, 0.0), MaxExhaustperCTPower(0.0),
-              DesignHeatRecVolFlowRate(0.0), DesignHeatRecMassFlowRate(0.0), DesignMinExitGasTemp(0.0), DesignAirInletTemp(0.0),
-              ExhaustStackTemp(0.0), HeatRecActive(false), HeatRecInletNodeNum(0), HeatRecOutletNodeNum(0), HeatRecInletTemp(0.0),
-              HeatRecOutletTemp(0.0), HeatRecMdot(0.0), HRPlantLoc{}, FuelMdot(0.0), FuelHeatingValue(0.0), ElecPowerGenerated(0.0),
-              ElecEnergyGenerated(0.0), HeatRecMaxTemp(0.0), OAInletNode(0), MyEnvrnFlag(true), MyPlantScanFlag(true), MySizeAndNodeInitFlag(true),
-              CheckEquipName(true), MyFlag(true)
+        CTGeneratorData() : HRPlantLoc{}
```
Comment (Myoldmopar) [inline src/EnergyPlus/CTElectricGenerator.hh:129]: This can go away.

File: src/EnergyPlus/ChilledCeilingPanelSimple.cc (hunk 12 above)
Comment (Myoldmopar) [reply src/EnergyPlus/ChilledCeilingPanelSimple.cc]: Yes, I suppose, and yes, this is a Fortran legacy thing.  Really, if the CheckEquipName needs to stay, it has no need to be its own separate array.  It should be a simple boolean on the chilled ceiling struct.  That's one better pattern we can propagate in this effort for sure.

File: src/EnergyPlus/ChilledCeilingPanelSimple.cc [hunk 20]
```diff
@@ -776,21 +775,21 @@ void InitCoolingPanel(EnergyPlusData &state, int const CoolingPanelNum, int cons
...
+        state.dataChilledCeilingPanelSimple->LastSysTimeElapsed.allocate((int)state.dataChilledCeilingPanelSimple->CoolingPanel.size());
         state.dataChilledCeilingPanelSimple->LastSysTimeElapsed = 0.0;
-        state.dataChilledCeilingPanelSimple->LastTimeStepSys.allocate(state.dataChilledCeilingPanelSimple->NumCoolingPanels);
+        state.dataChilledCeilingPanelSimple->LastTimeStepSys.allocate((int)state.dataChilledCeilingPanelSimple->CoolingPanel.size());
         state.dataChilledCeilingPanelSimple->LastTimeStepSys = 0.0;
-        state.dataChilledCeilingPanelSimple->SetLoopIndexFlag.allocate(state.dataChilledCeilingPanelSimple->NumCoolingPanels);
-        state.dataChilledCeilingPanelSimple->MySizeFlagCoolPanel.allocate(state.dataChilledCeilingPanelSimple->NumCoolingPanels);
+        state.dataChilledCeilingPanelSimple->SetLoopIndexFlag.allocate((int)state.dataChilledCeilingPanelSimple->CoolingPanel.size());
+        state.dataChilledCeilingPanelSimple->MySizeFlagCoolPanel.allocate((int)state.dataChilledCeilingPanelSimple->CoolingPanel.size());
```
Comment (Myoldmopar) [inline src/EnergyPlus/ChilledCeilingPanelSimple.cc]: All of these individual flag arrays being allocated to the same size screams that they should probably just be moved to scalars inside the chilled ceiling struct, reducing the number of individual arrays to maintain.  I understand there is a place where more arrays of scalars makes sense, but these are sporadically used flags scattered around multiple functions, so I don't believe this is that place.

File: src/EnergyPlus/ChilledCeilingPanelSimple.hh [hunk 21]
```diff
@@ -208,7 +194,6 @@ struct ChilledCeilingPanelSimpleData : BaseGlobalStruct
 
     bool GetInputFlag = true;
     bool MyOneTimeFlag = true;
-    int NumCoolingPanels = 0;
```
Comment (Myoldmopar) [inline src/EnergyPlus/ChilledCeilingPanelSimple.hh:211]: I think it's probably worth a little more time to get rid of a whole bunch of these Array1Ds.

File: src/EnergyPlus/ChillerAbsorption.hh [hunk 22]
```diff
@@ -223,15 +197,12 @@ namespace ChillerAbsorption {
...
-    int numAbsorbers = 0;
     bool getInput = true;
     Array1D<ChillerAbsorption::BLASTAbsorberSpecs> absorptionChillers;
 
     void clear_state() override
     {
-        this->numAbsorbers = 0;
-        this->getInput = true;
-        this->absorptionChillers.deallocate();
+        *this = ChillerAbsorberData();
```
Comment (Myoldmopar) [inline src/EnergyPlus/ChillerAbsorption.hh:205]: These *Data classes are starting to look a lot alike.  I think that's a good sign that things are getting normalized and will be opening up some refactoring next-steps sooner rather than later.

File: src/EnergyPlus/CostEstimateManager.hh [hunk 23]
```diff
@@ -200,14 +191,7 @@ struct CostEstimateManagerData : BaseGlobalStruct
 
     void clear_state() override
     {
-        this->GetCostInput = true;
-        this->NumLineItems = 0;
-        this->DoCostEstimate = false;
-        this->numMonetaryUnit = 0;
-        this->selectedMonetaryUnit = 0;
-        this->CurntBldg = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0};
```
Comment (Myoldmopar) [inline src/EnergyPlus/CostEstimateManager.hh:208]: You didn't do anything wrong here, but I dislike that the default values on the struct are all 0.0, but in both cases where we instantiate them, we pass a 1.0 as one of the arguments.  I think we should clean that up and just default assign it to 1.0 in the struct and then we don't have to have this initializer list anymore.

File: src/EnergyPlus/CurveManager.hh (hunk 3 above)
Comment (Myoldmopar) [reply src/EnergyPlus/CurveManager.hh:148]: :smile:

File: src/EnergyPlus/DataBranchNodeConnections.hh [hunk 24]
```diff
@@ -148,20 +131,7 @@ struct BranchNodeConnectionsData : BaseGlobalStruct
 
     void clear_state() override
     {
-        this->NumCompSets = 0;
-        this->NumNodeConnectionErrors = 0;
-        this->NumOfNodeConnections = 0;
-        this->MaxNumOfNodeConnections = 0;
-        this->NodeConnectionAlloc = 1000;
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataBranchNodeConnections.hh:155]: This variable smells funny.

File: src/EnergyPlus/DataEnvironment.hh [hunk 25]
```diff
@@ -208,123 +208,7 @@ struct EnvironmentData : BaseGlobalStruct
...
-        this->PrintEnvrnStampWarmupPrinted = false;
-        this->RunPeriodEnvironment = false;
-        this->EnvironmentStartEnd.clear();
-        this->CurrentYearIsLeapYear = false;
-        this->varyingLocationSchedIndexLat = 0;
-        this->varyingLocationSchedIndexLong = 0;
-        this->varyingOrientationSchedIndex = 0;
-        this->forceBeginEnvResetSuppress = false;
-        this->oneTimeCompRptHeaderFlag = true;
+        *this = EnvironmentData();
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataEnvironment.hh:211]: Yay!!!!!!!!

File: src/EnergyPlus/DataGlobals.hh [hunk 26]
```diff
@@ -175,112 +175,7 @@ struct DataGlobal : BaseGlobalStruct
...
-        this->errorCallback = nullptr;
-        this->eplusRunningViaAPI = false;
-        this->NumOfWaterHeater = 0;
-        this->CountNonZoneEquip = true;
-        this->FDsimDay = 0;
-        this->FDnumIterYears = 0;
-        this->printConsoleOutput = true;
-        this->installRootOverride = false;
-        this->numThread = 1;
+        *this = DataGlobal();
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataGlobals.hh:178]: Woo!!

**Review by Myoldmopar — COMMENTED** [review]
OK, so a net of 2400 lines removed.  No diffs.  Awesome.  This is making a bunch of big improvements in the headers and state classes.  I think it could use another pass based on my comments and some comments from @amirroth.  And then we can take the outcomes of this branch and apply them to the next group of files shortly.  Thanks @brianlball !

File: src/EnergyPlus/ChilledCeilingPanelSimple.cc (hunk 20 above)
Comment (amirroth) [reply src/EnergyPlus/ChilledCeilingPanelSimple.cc]: 📌  Yes, I am doing a 180 on the whole array-of-scalar thing. Or rather not a full 180, but a much narrower version of it coupled with a push in the other direction in most places.

File: src/EnergyPlus/Boilers.hh (hunk 1 above)
Comment (amirroth) [reply src/EnergyPlus/Boilers.hh:210]: So ... what is happening to the old instance? Does it get properly destroyed or are we leaking insane amounts of memory here? I admit I don't have a lot of experience with the `*this =` idiom. It looks very strange and wrong to me.

File: src/EnergyPlus/CostEstimateManager.hh (hunk 23 above)
Comment (brianlball) [reply src/EnergyPlus/CostEstimateManager.hh:208]: i think the default values have the same value as the reset now

File: src/EnergyPlus/CTElectricGenerator.hh (hunk 19 above)
Comment (brianlball) [reply src/EnergyPlus/CTElectricGenerator.hh:129]: should be gone now

File: src/EnergyPlus/Boilers.hh (hunk 16 above)
Comment (brianlball) [reply src/EnergyPlus/Boilers.hh:142]: should be gone now.

File: src/EnergyPlus/BoilerSteam.hh (hunk 7 above)
Comment (brianlball) [reply src/EnergyPlus/BoilerSteam.hh]: this was part of a diff debug and has been removed

File: src/EnergyPlus/Boilers.hh (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/Boilers.hh:210]: This should not be leaking anything.  I made a prototype of this and called `clear_state` 50bn times and the memory usage on the process never went up, and I could see the destructor being called every time, so the assignment seems to be doing everything we need here.  This paradigm was suggested by @lefticus, so I feel pretty good about it.

Now, that said, I think once we have all the `clear_state` functions cleaned up to this level, we can just go back to where we are calling `clear_state`, and change the call so that we don't need the intermediary `clear_state` function, if in fact all it is doing is deleting the last one and creating a new one.

(As a final aside, just note that this function is only called during serial unit test runs, and in case someone is calling clear_state during serial runs of E+ via API)

File: src/EnergyPlus/DataBranchNodeConnections.hh (hunk 24 above)
Comment (Myoldmopar) [reply src/EnergyPlus/DataBranchNodeConnections.hh:155]: It smells funny for two reasons:
1. We are carefully allocating out nodes during initialization in chunks, and I think we can rely on C++ to be smart enough to do this behind-the-scenes reservation.  Anyway, the second reason is that
2. This variable is only used in one function, so I'm moving it to a constexpr int inside that function.

File: src/EnergyPlus/DataContaminantBalance.hh [hunk 27]
```diff
@@ -288,8 +231,7 @@ struct ContaminantBalanceData : BaseGlobalStruct
 
     Array1D<Real64> CONTRAT;           // Zone CO2 at the previous time step used in Exact and Euler method
     Array1D<Real64> MixingMassFlowCO2; // Mixing MASS FLOW * CO2
-    int NumContControlledZones = 0;
-    Real64 OutdoorCO2 = 0.0; // Outdoor CO2 level
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataContaminantBalance.hh:292]: I'd love to go through this and clean up all the Array1D's, but I'm not going to add that to this PR.

File: src/EnergyPlus/DataLoopNode.hh [hunk 28]
```diff
@@ -1206,62 +1175,7 @@ struct LoopNodeData : BaseGlobalStruct
...
-                                                         0.0,
-                                                         0.0,
-                                                         0.0,
-                                                         0.0,
-                                                         false,
-                                                         false);
-        this->MoreNodeInfo.deallocate();
-        this->MarkedNode.deallocate();
-        this->NodeSetpointCheck.deallocate();
+        *this = LoopNodeData();
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataLoopNode.hh:1178]: :smile:

**Comment by Myoldmopar** [comment]
Just pushed a few commits that should get this ready to merge.  This PR cleans up so much, assuming it's clean let's merge this in.  If anyone has any remaining comments, speak up soon.

**Comment by Myoldmopar** [comment]
@amirroth thanks for commenting on this.  I responded, and am happy to keep pushing on this as needed to resolve them, but if some of them are more just notes to :pushpin: for later, I'll just carry on to get this merged in.

**Comment by Myoldmopar** [comment]
Done, I think this should be ready to drop if clean then.

**Comment by Myoldmopar** [comment]
All green, merging this one in.  Thanks @brianlball for the changes, and @amirroth for the commentary.

**Comment by amirroth** [comment]
Whatever the runtime implications of initializing in the object declaration, it's much more readable and understandable (given that you have the comments right there too) than having all of those in a giant constructor blob. Definitely a good development in terms of code readability and developer/reviewer sanity! Thanks!
